More Files plugin fixes

lukechampine opened the pull request:
Single-file downloads were not being downloaded to the correct location, since the download destination was constructed as: `${destination}/${file.name}`. For multiple files this makes sense, but for a single file it
constructs the wrong path; we just want `${destination}`.

Also replaced one use of `/` with `path.join`, to create proper paths on Windows. There may be some other uses of `/` that cause problems, but it seems unlikely; node apparently interprets `/` as the path separator regardless of operating system.

lukechampine committed:
fix downloading bug

single-file downloads were not being downloaded to the correct
location, since the download destination was constructed as:
  ${destination}/${file.name}
For multiple files this makes sense, but for a single file it
constructs the wrong path; we just want ${destination}.

Also replaced '/' with path.join, to create proper paths on Windows.

Mingling94 commented on plugins/Files/js/browser.js:
```diff
@@ -420,20 +427,17 @@ var browser = {
+			}
+			// Download each of the files
+			tools.notify(`Downloading ${label} to ${destination}`, 'download');
+				let functs = files.map(file => file.download.bind(file));
```
Tabbing, might as well fix it.

lukechampine committed:
Revert "append a / to folder names"

This reverts commit 80df8e7cc7f4bc21936c9667c67103fb215c5708.

Appending a / breaks other fragile behavior.

lukechampine committed:
disable downloading folders

this behavior requires more thought. For example, who should be in
charge of create the directories of a deeply-nested folder? If you
select a folder to download into, should you download the folder
there, or the folder's *contents* there? etc.

DavidVorick commented:
I am in favor of using `/` everywhere, even if node reinterprets it correctly on Windows. Experience tells me that being pedantic pays off when reliability is a goal.

DavidVorick commented:
Pull request looks good to me, but the tabbing should be fixed first.

lukechampine commented:
I don't understand. I think you made a typo.

Anyway, we probably shouldn't use `path.join` everywhere, since it means Windows users will create siapaths with `\` and others will create siapaths with `/`, and I don't know if that will cause incompatibility when sharing .sia files. It might be best to use `/` everywhere and convert the Windows `\` as soon as it's passed in.

Wait, hmm...if we use `/` internally, will it break siac on Windows? Calling `siac renter delete foo/bar/baz` on Windows might cause a parsing error. Ugh.

lukechampine committed:
fix uploadFolder bug

lukechampine committed:
always close dropdown after selection

DavidVorick commented:
Yeah I meant to say `path.Join` everywhere.

Urg and what happens when you share files across filesystems? Internally, `Sia` should definitely use `/` and then switch to the compatible version when going to disk.

hmph

DavidVorick commented on plugins/Files/js/browser.js:
```diff
@@ -61,6 +61,7 @@ function getSelectedFiles() {
 	var selected = $('.selected.file');
 	var nameFields = selected.map((i, el) => $(el).find('.name')).get();
 	var names = nameFields.map(field => field.text());
+	console.log(names);
```
did you mean to leave this here?

lukechampine replied:
damn

lukechampine commented:
at any rate, this PR is good to go. We can address these things more thoroughly in 0.5.1

DavidVorick commented on plugins/Files/js/browser.js:
```diff
@@ -375,7 +376,7 @@ var browser = {
 			tools.notify(`Sharing ${label} to ${destination}`, 'siafile');
 			loader.shareDotSia(paths, destination, function() {
-				tools.notify(`Put ${label}'s .sia files at ${destination}`, 'success');
+				tools.notify(`Put ${label}s .sia files at ${destination}`, 'success');
```
is it correct to be removing the apostrophy? What sorts of strings does ${label} expand to?

lukechampine replied:
just lazy committing on my part.
I removed the apostrophe while coding because it messed up my syntax highlighting

Mingling94 commented on plugins/Files/js/global.js:
```diff
@@ -85,7 +85,7 @@ $('.dropdown .button').click(function() {
 			$('#paste-ascii').show('fast');
 			$('#paste-ascii input').focus();
-			return; // Don't close dropdown
+			break;
```
The return here was needed to not close the dropdown.

lukechampine replied:
whoops :I
thanks for popping in to help us debug. We have our hands full at the moment.

Mingling94 commented on plugins/Files/js/global.js:
```diff
@@ -94,20 +94,17 @@ $('.dropdown .button').click(function() {
-
-	// Detect flawed userInput from actions that require it, hide if fine
-	if (!userInput && option !== 'Make Folder') {
-		tools.tooltip('Invalid action!', this);
```
If no input, silently fail and close the dropdown? Is that better?
